service/history/decisionTaskHandler.go

```diff
@@ -530,10 +552,10 @@ func (handler *decisionTaskHandlerImpl) handleDecisionRequestCancelExternalWorkf
 	}
 
 	cancelRequestID := uuid.New()
-	wfCancelReqEvent, _ := handler.mutableState.AddRequestCancelExternalWorkflowExecutionInitiatedEvent(
+	wfCancelReqEvent, _, err := handler.mutableState.AddRequestCancelExternalWorkflowExecutionInitiatedEvent(
 		handler.decisionTaskCompletedID, cancelRequestID, attr,
 	)
-	if wfCancelReqEvent == nil {
+	if err != nil {
 		return &workflow.InternalServiceError{Message: "Unable to add external cancel workflow request."}
```

Comment:
Append error in the message?

Reply:
i will create another PR deal with this.
this PR mainly focus on adding error, not changing the errors returned